src/gui/PasswordEdit.cpp

```diff
@@ -119,3 +142,81 @@ void PasswordEdit::autocompletePassword(const QString& password)
         setText(password);
     }
 }
+
+bool PasswordEdit::event(QEvent* event)
+{
+    if (!m_basePasswordEdit) {
+        // check caps lock state only for primary password edits
+        checkCapslockState();
+    }
+    return QLineEdit::event(event);
+}
+
+#include <iostream>
+void PasswordEdit::checkCapslockState()
+{
+    bool newCapslockState = m_capslockState;
+    Q_UNUSED(m_capslockState)
```

Comment:
Necessary?

Reply:
I think it was at some point, because not all ifdef branches included this, but it seems like it can be removed now.